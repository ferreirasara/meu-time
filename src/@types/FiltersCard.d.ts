export type FiltersCardProps = {
  selectedCountry: string
  handleSelectCountry: (country: string) => void
  selectedLeague: number
  handleSelectLeague: (league: number) => void
  selectedSeason: number
  handleSelectSeason: (season: number) => void
  selectedTeam: number
  handleSelectTeam: (team: number) => void
}