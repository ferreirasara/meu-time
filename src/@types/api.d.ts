export type Team = {
  team: {
    id: number,
    name: string,
    code: string,
    country: string,
    founded: number,
    national: boolean,
    logo: string,
  },
  venue: {
    id: number,
    name: string,
    address: string,
    city: string,
    capacity: number,
    surface: string,
    image: string,
  },
}

export type Country = {
  name: string,
  code?: string,
  flag?: string,
}

export type League = {
  country: Country,
  league: {
    id: number,
    name: string,
    type: string,
    logo: string,
  }
  seasons: {
    year: number,
    start: string,
    end: string,
    current: boolean,
    coverage: {
      fixtures: {
        events: boolean,
        lineups: boolean,
        statistics_fixtures: boolean,
        statistics_players: boolean
      },
      standings: boolean,
      players: boolean,
      top_scorers: boolean,
      top_assists: boolean,
      top_cards: boolean,
      injuries: boolean,
      predictions: boolean,
      odds: boolean
    }
  }[],
}