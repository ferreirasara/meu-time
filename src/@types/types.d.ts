export type BaseData = {
  lastUpdated?: Date,
  countries?: Country[],
  leagues?: League[],
  seasons?: number[],
}

export type UserData = {
  email: string,
  firstname: string,
  lastname: string,
}

export type RequestData = {
  current: number,
  limitDay: number,
}

export type PlayersTableData = {
  id: number,
  name: string,
  age: number,
  nationality: string,
  photo: string,
}

export type PlayersTableProps = {
  selectedLeague: number
  selectedSeason: number
  selectedTeam: number
  handleError: (message: string) => void
}

export type GoalsChartProps = {
  data?: GeneralStatsData,
  loading: boolean,
}

export type GeneralStatsData = {
  totalDraws: number,
  totalPlayed: number,
  totalWins: number,
  totalLoses: number,
  mostUsedFormation: {
    formation: string,
    played: number,
  },
  goals: {
    against: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
    for: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
  }
}

export type GeneralStatsProps = {
  selectedLeague: number
  selectedSeason: number
  selectedTeam: number
  handleError: (message: string) => void
}

export type FiltersCardProps = {
  selectedCountry: string
  handleSelectCountry: (country: string) => void
  selectedLeague: number
  handleSelectLeague: (league: number) => void
  selectedSeason: number
  handleSelectSeason: (season: number) => void
  selectedTeam: number
  handleSelectTeam: (team: number) => void
  localStorageDataLoading: boolean
  handleError: (message: string) => void
}

export type Team = {
  team: {
    id: number,
    name: string,
    code: string,
    country: string,
    founded: number,
    national: boolean,
    logo: string,
  },
  venue: {
    id: number,
    name: string,
    address: string,
    city: string,
    capacity: number,
    surface: string,
    image: string,
  },
}

export type Country = {
  name: string,
  code?: string,
  flag?: string,
}

export type League = {
  country: Country,
  league: {
    id: number,
    name: string,
    type: string,
    logo: string,
  }
  seasons: {
    year: number,
    start: string,
    end: string,
    current: boolean,
    coverage: {
      fixtures: {
        events: boolean,
        lineups: boolean,
        statistics_fixtures: boolean,
        statistics_players: boolean
      },
      standings: boolean,
      players: boolean,
      top_scorers: boolean,
      top_assists: boolean,
      top_cards: boolean,
      injuries: boolean,
      predictions: boolean,
      odds: boolean
    }
  }[],
}

export type Player = {
  player: {
    id: number,
    name: string,
    firstname: string,
    lastname: string,
    age: number,
    birth: {
      date: string,
      place: string,
      country: string,
    },
    nationality: string,
    height: string,
    weight: string,
    injured: boolean,
    photo: string,
  },
  // statistics: {
  //   "team": {
  //     "id": 147,
  //     "name": "Coritiba",
  //     "logo": "https://media-1.api-sports.io/football/teams/147.png"
  //   },
  //   "league": {
  //     "id": 71,
  //     "name": "Serie A",
  //     "country": "Brazil",
  //     "logo": "https://media-2.api-sports.io/football/leagues/71.png",
  //     "flag": "https://media-3.api-sports.io/flags/br.svg",
  //     "season": 2016
  //   },
  //   "games": {
  //     "appearences": 6,
  //     "lineups": 6,
  //     "minutes": 438,
  //     "number": null,
  //     "position": "Defender",
  //     "rating": null,
  //     "captain": false
  //   },
  //   "substitutes": {
  //     "in": 0,
  //     "out": 3,
  //     "bench": 0
  //   },
  //   "shots": {
  //     "total": null,
  //     "on": null
  //   },
  //   "goals": {
  //     "total": 0,
  //     "conceded": null,
  //     "assists": null,
  //     "saves": null
  //   },
  //   "passes": {
  //     "total": null,
  //     "key": null,
  //     "accuracy": null
  //   },
  //   "tackles": {
  //     "total": null,
  //     "blocks": null,
  //     "interceptions": null
  //   },
  //   "duels": {
  //     "total": null,
  //     "won": null
  //   },
  //   "dribbles": {
  //     "attempts": null,
  //     "success": null,
  //     "past": null
  //   },
  //   "fouls": {
  //     "drawn": null,
  //     "committed": null
  //   },
  //   "cards": {
  //     "yellow": 0,
  //     "yellowred": 0,
  //     "red": 0
  //   },
  //   "penalty": {
  //     "won": null,
  //     "commited": null,
  //     "scored": null,
  //     "missed": null,
  //     "saved": null
  //   }
  // }[]
}

export type TeamStats = {
  lineups: {
    formation: string,
    played: number
  }[],
  fixtures: {
    draws: { home: number, away: number, total: number }
    loses: { home: number, away: number, total: number },
    played: { home: number, away: number, total: number },
    wins: { home: number, away: number, total: number },
  },
  goals: {
    against: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
    for: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
  }
}

export type InfoModalProps = {
  infoModalOpen: boolean
  onClose: () => void
}