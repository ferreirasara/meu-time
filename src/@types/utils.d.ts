export type BaseData = {
  lastUpdated?: Date,
  countries?: Country[],
  leagues?: League[],
  seasons?: number[],
}