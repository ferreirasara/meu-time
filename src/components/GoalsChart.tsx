import { Line } from "react-chartjs-2"
import { Chart as ChartJS, CategoryScale, LinearScale, PointElement, LineElement, Title, Tooltip, Legend } from 'chart.js';
import { Divider } from "antd";
import { GoalsChartProps } from "../@types/types";

ChartJS.register(CategoryScale, LinearScale, PointElement, LineElement, Title, Tooltip, Legend);

export const GoalsChart = ({ data, loading }: GoalsChartProps) => {
  const options = {
    responsive: true,
  };

  const labels = ["0-15", "16-30", "31-45", "46-60", "61-75", "76-90", "91-105", "106-120"];

  const chartData = {
    labels,
    datasets: [
      {
        label: 'Gols feitos',
        data: [
          data?.goals?.for?.minute?.["0-15"]?.total,
          data?.goals?.for?.minute?.["16-30"]?.total,
          data?.goals?.for?.minute?.["31-45"]?.total,
          data?.goals?.for?.minute?.["46-60"]?.total,
          data?.goals?.for?.minute?.["61-75"]?.total,
          data?.goals?.for?.minute?.["76-90"]?.total,
          data?.goals?.for?.minute?.["91-105"]?.total,
          data?.goals?.for?.minute?.["106-120"]?.total,
        ],
        borderColor: 'rgb(53, 162, 235)',
        backgroundColor: 'rgba(53, 162, 235, 0.5)',
      },
      {
        label: 'Gols levados',
        data: [
          data?.goals?.against?.minute?.["0-15"]?.total,
          data?.goals?.against?.minute?.["16-30"]?.total,
          data?.goals?.against?.minute?.["31-45"]?.total,
          data?.goals?.against?.minute?.["46-60"]?.total,
          data?.goals?.against?.minute?.["61-75"]?.total,
          data?.goals?.against?.minute?.["76-90"]?.total,
          data?.goals?.against?.minute?.["91-105"]?.total,
          data?.goals?.against?.minute?.["106-120"]?.total,
        ],
        borderColor: 'rgb(255, 99, 132)',
        backgroundColor: 'rgba(255, 99, 132, 0.5)',
      },
    ],
  };

  return <div>
    <Divider>Gols por minuto de jogo</Divider>
    <Line options={options} data={chartData} data-testid="goals-chart" />
  </div>
}