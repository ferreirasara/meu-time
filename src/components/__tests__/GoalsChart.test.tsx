import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { GoalsChart } from '../GoalsChart';
import { GoalsChartProps } from '../../@types/types';

const props: GoalsChartProps = {
  data: {
    goals: {
      against: {
        minute: {
          "0-15": { percentage: "0%", total: 0 },
          "16-30": { percentage: "0%", total: 0 },
          "31-45": { percentage: "0%", total: 0 },
          "46-60": { percentage: "0%", total: 0 },
          "61-75": { percentage: "0%", total: 0 },
          "76-90": { percentage: "0%", total: 0 },
          "91-105": { percentage: "0%", total: 0 },
          "106-120": { percentage: "0%", total: 0 },
        }
      },
      for: {
        minute: {
          "0-15": { percentage: "0%", total: 0 },
          "16-30": { percentage: "0%", total: 0 },
          "31-45": { percentage: "0%", total: 0 },
          "46-60": { percentage: "0%", total: 0 },
          "61-75": { percentage: "0%", total: 0 },
          "76-90": { percentage: "0%", total: 0 },
          "91-105": { percentage: "0%", total: 0 },
          "106-120": { percentage: "0%", total: 0 },
        }
      },
    },
    mostUsedFormation: { formation: "", played: 0 },
    totalDraws: 0,
    totalLoses: 0,
    totalPlayed: 0,
    totalWins: 0,
  },
  loading: false,
}

describe('GoalsChart', () => {
  beforeAll(() => {
    window.matchMedia = (query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })
  })

  it('match snapshot', () => {
    const goalsChart = renderer.create(<GoalsChart {...props} />).toJSON();
    expect(goalsChart).toMatchSnapshot();
  });

  it('renders canvas chart', () => {
    const { getByTestId } = render(<GoalsChart {...props} />);

    const chartEl = getByTestId('goals-chart');
    expect(chartEl).toBeInTheDocument();
  });
})