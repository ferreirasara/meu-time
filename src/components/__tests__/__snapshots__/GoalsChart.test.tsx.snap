// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalsChart match snapshot 1`] = `
<div>
  <div
    className="ant-divider css-dev-only-do-not-override-15rg2km ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center"
    role="separator"
  >
    <span
      className="ant-divider-inner-text"
      style={Object {}}
    >
      Gols por minuto de jogo
    </span>
  </div>
  <canvas
    data-testid="goals-chart"
    height={150}
    role="img"
    width={300}
  />
</div>
`;
